solara/components/togglebuttons.py

```diff
@@ -77,35 +109,75 @@ def Page():
     """
     class_ = _combine_classes(classes)
     style_flat = solara.util._flatten_style(style)
     # TODO: make type safe
+    # typing is ignored below due to an issue with the typing of use_reactive; Typechecker expects a different type than is given. If expected type is
+    # given instead, a similar error shows up in typing of argument.
     reactive_value = solara.use_reactive(value, on_value)  # type: ignore
     children = [solara.Button(label=str(value)) for value in values] + children
     values = values + [_get_button_value(button) for button in children]  # type: ignore
-    index, set_index = solara.use_state_or_update(values.index(reactive_value.value) if reactive_value.value is not None else 0, key="index")
+    # When mandatory = True, index should not be None, but we are letting the front-end take care of setting index to 0 because of a bug
+    # (see https://github.com/widgetti/solara/issues/282)
+    # TODO: set index to 0 on python side (after #282 is resolved)
+    index, set_index = solara.use_state_or_update(values.index(reactive_value.value) if reactive_value.value is not None else None, key="index")
 
     def on_index(index):
         set_index(index)
-        value = values[index]
+        if mandatory:
+            value = values[index]
+        else:
+            value = values[index] if index is not None else None
         reactive_value.set(value)
 
     return cast(
         reacton.core.ValueElement[v.BtnToggle, T],
-        rv.BtnToggle(children=children, multiple=False, mandatory=True, v_model=index, on_v_model=on_index, dense=dense, class_=class_, style_=style_flat),
+        rv.BtnToggle(children=children, multiple=False, mandatory=mandatory, v_model=index, on_v_model=on_index, dense=dense, class_=class_, style_=style_flat),
     )
 
 
+@overload
 @solara.value_component(None)
 def ToggleButtonsMultiple(
     value: Union[List[T], solara.Reactive[List[T]]],
+    values: List[T] = ...,
+    children: List[reacton.core.Element] = ...,
+    on_value: Union[Callable[[List[T]], None], None] = ...,
+    dense: bool = ...,
+    mandatory: Literal[True] = ...,
+    classes: List[str] = ...,
+    style: Union[str, Dict[str, str], None] = ...,
+) -> reacton.core.ValueElement[v.BtnToggle, List[T]]:
+    ...
+
+
+@overload
+@solara.value_component(None)
+def ToggleButtonsMultiple(
+    value: Union[Optional[List[T]], solara.Reactive[Optional[List[T]]]] = [],
+    values: List[T] = ...,
+    children: List[reacton.core.Element] = ...,
+    on_value: Union[Callable[[List[T]], None], None] = ...,
+    dense: bool = ...,
+    mandatory: Literal[False] = ...,
+    classes: List[str] = ...,
+    style: Union[str, Dict[str, str], None] = ...,
+) -> reacton.core.ValueElement[v.BtnToggle, List[T]]:
+    ...
+
+
+@solara.value_component(None)
+def ToggleButtonsMultiple(
+    value: Union[Optional[List[T]], solara.Reactive[List[T]], solara.Reactive[Optional[List[T]]]] = [],
```

Comment:
I think there is nothing to change/overload here. I don't think we should do Optional[List[T]], and allow None, but only use an empty list to encode nothing is selected. The only things that should be constrained when mandatory=True is that the length of the list should be of length 1, which we cannot express in the typing system (it is a runtime thing).

Reply:
So just to be clear, remove all overloading from ToggleButtonsMultiple?

Reply:
Yes, unless I am mistaken, the typing does not depend on the boolean argument `mandatory` anymore.

Reply:
Yeah, that's correct when we don't allow for `value` to be `None`

Reply:
Regardless, the code still works for `value=None`, but I don't think we want to do anything about that.